.gitlab-ci.yml

```diff
@@ -1,118 +1,142 @@
   services:
     - docker:dind
-  before_script:
-  - mkdir -p /go/src/github.com/StackVista
-  - ln -s $CI_PROJECT_DIR  /go/src/github.com/StackVista/stackstate-agent
   script:
-  - cd /go/src/github.com/StackVista/stackstate-agent
-  - pip install -r requirements.txt
-  - inv deps
-  - inv -e docker.integration-tests
+    - inv -e docker.integration-tests
   tags:
-  - sts-aws
+    - sts-aws
 
 build_binaries:
-  stage: build
-  image: stackstate/stackstate-agent-runner-circle:latest
-  before_script:
-  - mkdir -p /go/src/github.com/StackVista
-  - ln -s $CI_PROJECT_DIR  /go/src/github.com/StackVista/stackstate-agent
+  stage: build_and_test
+  dependencies:
+    - deps
   script:
-  - cd /go/src/github.com/StackVista/stackstate-agent
-  - pip install -r requirements.txt
-  - inv deps
-  - inv -e dogstatsd.build --static
-  - inv -e agent.build
-
+    - inv -e dogstatsd.build --static
+    - inv -e agent.build
 
 build_puppy:
-  stage: build
-  image: stackstate/stackstate-agent-runner-circle:latest
-  before_script:
-  - mkdir -p /go/src/github.com/StackVista
-  - ln -s $CI_PROJECT_DIR  /go/src/github.com/StackVista/stackstate-agent
+  stage: build_and_test
+  dependencies:
+    - deps
   script:
-  - cd /go/src/github.com/StackVista/stackstate-agent
-  - pip install -r requirements.txt
-  - inv deps
-  - inv -e agent.build --puppy
-  - ./bin/agent/agent -c ./bin/agent/dist check cpu
-
-
-build_publish_deb:
-  stage: build
-  image: stackstate/stackstate-agent-runner-circle:latest
-  before_script:
-  - mkdir -p /go/src/github.com/StackVista
-  - ln -s $CI_PROJECT_DIR  /go/src/github.com/StackVista/stackstate-agent
+    - inv -e agent.build --puppy
+    - ./bin/agent/agent -c ./bin/agent/dist check cpu
+
+build_deb_clear_cache:
+  stage: build_and_test
+  cache:
+    policy: push
+    key: "agent-build_publish_deb"
+    paths:
+      - .omnibus
+      - vendor
+  when: manual
   script:
-  - printenv
-  - echo "======================================================"
-  - export CODE_ROOT=/go/src/github.com/StackVista/stackstate-agent
-    export OMNIBUS_ROOT=/omnibus
-  - cd $CODE_ROOT
-  - pip install -r requirements.txt
-  - inv deps
-  - git config --global user.email "circleci@stackstate"
-  - git config --global user.name "CircleCI Stackstate"
-  - apt-get install -y apt-utils libtool && apt-get install -y --only-upgrade libtool
-    # Drop symlink because it will fail the build when coming from a cache
-  - rm -f $OMNIBUS_ROOT/src/datadog-agent/src/github.com/DataDog/datadog-agent/vendor/github.com/coreos/etcd/cmd/etcd || true
-  - printenv
-  - inv -e agent.omnibus-build --base-dir $OMNIBUS_ROOT --skip-sign --release-version=6.5.2
+    - mkdir -p .omnibus
+    - rm -rf .omnibus/*
+    - rm -rf vendor
+
+build_deb:
+  stage: build_and_test
+  cache:
+    key: "agent-build_publish_deb"
+    paths:
+      - .omnibus
+      - vendor
+  script:
+    # We move .omnibus out of the CI root because this seems to cause issues with git
```

Comment:
extract this in an external script ?

Reply:
i find it contains too much giltab-related stuff (that would not be runnable in an external script). E.g. its all caching and stuff, i'd rather keep that here in one place